elalog/filewriter.go

```diff
@@ -41,64 +41,60 @@ func (w *fileWriter) writeHandler() {
 	var fileSize int64
 	var folderSize int64
 
-	for {
-		select {
-		case buf := <-w.writeChan:
-			var bufLen = int64(len(buf))
-
-			// create new log file if current file is nil or reach max size.
-			if atomic.AddInt64(&fileSize, bufLen) >= w.maxFileSize ||
-				current == nil {
-
-				// force create new log file
-				var err error
-				var file *os.File
-				for file, err = newLogFile(w.path); err != nil; {
-					fmt.Printf("New log file %s, err %v\n", w.path, err)
-					file, err = newLogFile(w.path)
-				}
-
-				// close previous log file in an other goroutine.
-				go func(file *os.File) {
-					if file != nil {
-						// force close file
-						for err := file.Close(); err != nil; {
-							err = file.Close()
-						}
-					}
-				}(current)
+	files, _ := ioutil.ReadDir(w.path)
+	for _, f := range files {
+		folderSize += f.Size()
+	}
```

Comment:
Each time the `fileWriter` starts, `folderSize` needs to be initialized. So I add these codes here.